packages/form-js-viewer/src/util/feel.js

```diff
@@ -0,0 +1,24 @@
+import { parseUnaryTests } from 'feelin';
+
+/**
+ * Retrieve variable names from given FEEL unary test.
+ *
+ * @param {string} unaryTest
+ * @returns {string[]}
+ */
+export function getVariableNames(unaryTest) {
+  const tree = parseUnaryTests(unaryTest);
+  const cursor = tree.cursor();
+
+  const variables = new Set();
+  do {
+    const node = cursor.node;
+
+    if (node.type.name === 'VariableName') {
+      variables.add(unaryTest.slice(node.from, node.to));
+    }
+
+  } while (cursor.next());
```

Comment:
Couldn't we also do `tree.iterate` here? Maybe it's safer? I don't know much about the insights.

```js
tree.iterate({
  enter: (node) => {
    if (node.name === 'VariableName') {
      variables.add(unaryTest.slice(node.from, node.to));
    }
  }  
});
```

Reply:
I tried both and liked cursor-based solution more. In `tree.iterate`, you have to know that iteration is done synchronously, while with cursor it's explicit.